django/contrib/gis/gdal/datasource.py

```diff
@@ -64,11 +63,9 @@ def __init__(self, ds_input, ds_driver=False, write=False, encoding="utf-8"):
         Driver.ensure_registered()
 
         if isinstance(ds_input, (str, Path)):
-            # The data source driver is a void pointer.
-            ds_driver = Driver.ptr_type()
             try:
-                # OGROpen will auto-detect the data source type.
-                ds = capi.open_ds(force_bytes(ds_input), self._write, byref(ds_driver))
+                # GDALOpenEx will auto-detect the data source type.
+                ds = capi.open_ds(force_bytes(ds_input), self._write, None, None, None)
```

Comment:
Stepping through `gis_tests.inspectapp.tests.OGRInspectTest.test_time_field` before this patch I see:
```
(Pdb) ds
c_void_p(2751523152928)
(Pdb) capi.get_layer_count(ds)
49
(Pdb) ds_input
"PG:dbname='test_geodjango' user='postgres' password='david'"
(Pdb) capi.get_ds_name(ds)
b"PG:dbname='test_geodjango' user='postgres' password='david'"
```

with this patch I see:

```
(Pdb) ds
c_void_p(2276731897056)
(Pdb) capi.get_layer_count(ds)
0
(Pdb) ds_input
"PG:dbname='test_geodjango' user='postgres' password='david'"
(Pdb) capi.get_ds_name(ds)
b"PG:dbname='test_geodjango' user='postgres' password='david'"
(Pdb)
```

It seems there is an issue with opening the datasource? 🤔